features/topology-map/org.opennms.features.topology.app/src/main/java/org/opennms/features/topology/app/internal/jung/D3TopoLayout.java

```diff
@@ -214,6 +214,10 @@ public boolean visitNode(Node<VertexData> n) {
     @Override
     public void step() {
 
+        if(getGraph().getEdges().isEmpty()){
+            m_alpha = 0; // => makes done() == true. We don't need to iterate over nothing
```

Comment:
We are still iterating at least once. Can't we just do this while initializing the layout and not invoke `step()` at all?
Can we also get rid of the `stepOld()` method if we are here anyways changing code?

Reply:
`stepOld()`=> removed
I am not sure I understand the rest of your suggestion. Where do you see where we can save time?

Reply:
Ah now I understand. I solved it a bit differently though. I moved that logic directly into the done method which I believe is the right place for it since that method is responsible for determining if the layout is finished or not. @mvrueden do you agree?

Reply:
Actually I think it is not that important. Both `done()` or `initialize()` are equally good. I don't prefer the one over the other 👍